### PR TITLE
Issue #4545: deprecate test only FileText constructor

### PR DESCRIPTION
**Pull request by rnveach**
Issue #4545

Constructor is only used by tests.
New issue should be created to fully remove this method.

### CODE REVIEW

**Comment by rnveach** [comment]
Issue is not approved so travis will fail.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/FileText.java [hunk 1]
```diff
@@ -166,7 +166,9 @@ public FileText(FileText fileText) {
      * @param file the name of the file
      * @param lines the lines of the text, without terminators
      * @throws NullPointerException if the lines array is null
+     * @deprecated Use {@link #FileText(File, String)} to load the file directly.
      */
+    @Deprecated
     public FileText(File file, List<String> lines) {
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/FileText.java]: Is the `file` parameter required here? It is always "filename" in the tests. Maybe there should be factory method `fromLines` or something like that.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/FileText.java]: It is a final field, we must give it something. we pass file from here to other places like `FileContents` which is used to give the file name to checks.
Tests were also just written using this as a way around actually storing a file, so I wouldn't put high emphasis on the tests and look to what is being done in the production code.
I do know at one time we were considering merging this file with `FileContents`.

**Comment by rnveach** [comment]
Closing to remove entire constructor like @romani suggested.

**Pull request closed without merging**